### PR TITLE
Add System.Runtime.CompilerServices.ReadOnlyAttribute

### PR DESCRIPTION
**Pull request by OmarTawfik**
Proposal is in: https://github.com/dotnet/corefx/issues/17741

Questions for area owners:

1) Is this the correct branch to use? is a rebase needed?
2) Do I need to run any additional tools? I see that `model.xml` is no longer used in the repo.
3) ~~Is there any additional work needed to copy this to corert repo? or is the mirroring tool going to handle it?~~

cc @VSadov @jcouv @dotnet/roslyn-compiler

### CODE REVIEW

File: src/mscorlib/src/System/Runtime/CompilerServices/ReadOnlyAttribute.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+
```
Comment (jaredpar) [inline src/mscorlib/src/System/Runtime/CompilerServices/ReadOnlyAttribute.cs]: Double blank line

File: src/mscorlib/src/System/Runtime/CompilerServices/ReadOnlyAttribute.cs [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+using System;
+
+namespace System.Runtime.CompilerServices
+{
+    [AttributeUsage(AttributeTargets.All, Inherited = false)]
```
Comment (jaredpar) [inline src/mscorlib/src/System/Runtime/CompilerServices/ReadOnlyAttribute.cs]: Should add  doc comment detailing this is an attribute used by the compiler to trakc additional metadata.  Should not be used by developers manually.

File: src/mscorlib/src/System/Runtime/CompilerServices/ReadOnlyAttribute.cs (hunk 1 above)
Comment (OmarTawfik) [inline src/mscorlib/src/System/Runtime/CompilerServices/ReadOnlyAttribute.cs]: Convention of other files in the same folder :(

File: src/mscorlib/src/System/Runtime/CompilerServices/ReadOnlyAttribute.cs (hunk 2 above)
Comment (jcouv) [inline src/mscorlib/src/System/Runtime/CompilerServices/ReadOnlyAttribute.cs]: Maybe mark type with `EditorBrowsableAttribute(EditorBrowsableState.Never)`?
On the other hand, I think the entire CompilerServices namespace is already meant for compiler consumption (but not general consumption), so may be superfluous.

Comment (stephentoub) [inline src/mscorlib/src/System/Runtime/CompilerServices/ReadOnlyAttribute.cs]: > I think the entire CompilerServices namespace is already meant for compiler consumption (but not general consumption), so may be superfluous

There are types in the compiler namespace that folks use, though, like ConditionalWeakTable, StrongBox, even AsyncTaskMethodBuilder.  I think adding the EditorBrowsable attribute would be reasonable.

Comment (OmarTawfik) [inline src/mscorlib/src/System/Runtime/CompilerServices/ReadOnlyAttribute.cs]: My plan for now is to produce errors when users use it in their source code, same as we do for `Dynamic` attribute anyways. Will add the comment.

**Comment by OmarTawfik** [comment]
@terrajobst @karelz do I need sign off from API before merging in?

**Comment by jkotas** [comment]
The API issues is marked as approved. There is no additional sign off required (except regular PR review).

**Pull request merged**